File: ecommerce-backend/src/main/java/com/ecommerce/ecommerce_backend/entity/Product.java
package com.ecommerce.ecommerce_backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private BigDecimal price;

    @Column(name = "sale_price")
    private BigDecimal salePrice;

    @Column(nullable = false)
    private Integer stock;

    @Column(unique = true)
    private String sku;

    private String thumbnail;

    @Enumerated(EnumType.STRING)
    private ProductStatus status;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        status = ProductStatus.ACTIVE;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum ProductStatus {
        ACTIVE,
        OUT_OF_STOCK,
        HIDDEN
    }
}

File: ecommerce-mobile/app/src/main/java/android/app/ecommerce/data/model/Product.kt
package android.app.ecommerce.data.model


data class Product(
    val id: Long,
    val sellerId: Long,
    val categoryId: Long? = null,
    val name: String,
    val description: String? = null,
    val price: Double,
    val salePrice: Double? = null,
    val stock: Int,
    val sku: String? = null,
    val thumbnail: String? = null,
    val status: ProductStatus,
    val createdAt: String? = null,  // hoặc LocalDateTime nếu xử lý JSON tốt
    val updatedAt: String? = null
) {
    enum class ProductStatus {
        ACTIVE,
        OUT_OF_STOCK,
        HIDDEN
    }
}